modified and finalized Cookie-Based Authentication and Role-Based Dashboard Redirection

YuejiaoShi opened the pull request.

YuejiaoShi committed:
add js-cookie and jsonwebtoken

YuejiaoShi committed:
modified and finalized Cookie-Based Authentication and Role-Based Dashboard Redirection

YuejiaoShi committed:
remove typescript dependencies

YuejiaoShi committed:
reimport files

YuejiaoShi committed:
replace jsonwebtoken with jose for JWT verification and decoding

magdazelena commented on frontend/app/_components/RoleSelection.js:
```diff
@@ -6,14 +6,14 @@ const RoleSelection = ({ setRole }) => {
 
       <button
-        onClick={() => setRole('developer')}
+        onClick={() => setRole('Developer')}
```
considering the fact that you have two roles set as a string and don't use Typescript, I would suggest making an enum with the roles. This way you will ensure that there are not bugs related to typos in string or upper/lower case missmatch etc.

magdazelena commented on frontend/app/layout.js:
```diff
@@ -1,34 +1,29 @@
-  src: "./fonts/GeistVF.woff",
-  variable: "--font-geist-sans",
-  weight: "100 900",
+  src: './fonts/GeistVF.woff',
```
you folks reaaaaaally need a eslint in your project config that is shared in repo, this is one of many times I see change of quotes in the diff and this is due to the fact that you don't have streamlined config as the team! Please look into this, this is part of good practice :)

magdazelena commented on frontend/app/utils/auth.js:
```diff
@@ -1,49 +1,58 @@
+export const handleSignUp = async data => {
   try {
-    const response = await fetch('/api/users', {
+    const response = await fetch(`/api/users`, {
```
there is no need to use string literal if no variables are injected :)

magdazelena commented on frontend/app/utils/auth.js:
```diff
@@ -1,49 +1,58 @@
+
     if (response.ok) {
-      return { success: true, role: data.role };
+      alert(result.message + ' Sign In again.');
```
here, on the other hand, is a great usecase to use string literal. It's generally better practice than concatenating with `+` :)

magdazelena reviewed (changes requested):
Logic-wise is seems good, great work! I have some issues though with formatting and semantics, as well as using string as reference :) I put them in one place but of course they are applicable to all occurences :)

YuejiaoShi committed:
Use /api/user to get user info instead of decode token

1. refactor post and get request functions
2. install toastify notification
3. remove jwt verify
4. not decode token anymore

YuejiaoShi committed:
downgrade react to 18

YuejiaoShi committed:
remove redirect when token is empty

YuejiaoShi committed:
add checkUserRole logic
